internal/wire/parse.go

```diff
@@ -845,6 +872,62 @@ func processInterfaceValue(fset *token.FileSet, info *types.Info, call *ast.Call
 	}, nil
 }
 
+// processFieldsOf creates a list of fields from a wire.FieldsOf call.
+func processFieldsOf(fset *token.FileSet, info *types.Info, call *ast.CallExpr) ([]*Field, error) {
+	// Assumes that call.Fun is wire.FieldsOf.
+
+	if len(call.Args) < 2 {
+		return nil, notePosition(fset.Position(call.Pos()),
+			errors.New("call to FieldsOf must specify fields to be extracted"))
+	}
+	structType := info.TypeOf(call.Args[0])
+	structPtr, ok := structType.(*types.Pointer)
+	if !ok {
+		return nil, notePosition(fset.Position(call.Pos()),
+			fmt.Errorf("first argument to FieldsOf must be a pointer to a struct; found %s", types.TypeString(structType, nil)))
+	}
+	struc, ok := structPtr.Elem().Underlying().(*types.Struct)
```

Comment:
Shouldn't we also permit `**S` (written as `wire.FieldsOf(new(*S), ...)`) for getting the field from a `*S`?

Reply:
The latter. The generated code should be the same, but the input/parent type in the provider graph would be different (`*S` instead of `S`).